### PR TITLE
Use builder to construct RoutineCharacteristics

### PR DESCRIPTION
**Pull request by caithagoras**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/function/RoutineCharacteristics.java [hunk 1]
```diff
@@ -68,6 +80,16 @@ public boolean isCalledOnNullInput()
         return nullCallClause == CALLED_ON_NULL_INPUT;
     }
 
+    public Builder toBuilder()
```
Comment (rongrong) [inline presto-spi/src/main/java/com/facebook/presto/spi/function/RoutineCharacteristics.java]: This is a little strange. It sounds like converting an object to a builder, where you actually mean creating a builder from this object. I prefer `builder().from(..)` or just `builder(...)`.

Comment (caithagoras) [reply presto-spi/src/main/java/com/facebook/presto/spi/function/RoutineCharacteristics.java]: Done

File: presto-spi/src/main/java/com/facebook/presto/spi/function/RoutineCharacteristics.java [hunk 2]
```diff
@@ -94,4 +116,45 @@ public String toString()
...
         return format("(%s, %s, %s)", language, determinism, nullCallClause);
     }
+
+    public static class Builder
+    {
+        private Language language = SQL;
+        private Determinism determinism = NOT_DETERMINISTIC;
+        private NullCallClause nullCallClause = CALLED_ON_NULL_INPUT;
+
+        private Builder()
```
Comment (rongrong) [inline presto-spi/src/main/java/com/facebook/presto/spi/function/RoutineCharacteristics.java]: I think we normally do empty constructor in a single line
`private Builder() {}`

**Comment by caithagoras** [comment]
@rongrong comments addressed.

**Pull request merged**